### DISCUSSION TITLE
`NumberInput` should not remove 0 value

### DISCUSSION THREAD

**Discussion by rimzzlabs** [mantinedev/mantine#4781] [Q&A]
How to prevent `0` value from being removed when using `<NumberInput/>` ?

**Comment by rtivital** [comment]
You can use controlled component and if the `onChange` function is called with an empty string, set state with `0` instead.

**Reply by rimzzlabs** [reply]
Hi thank you for your reply

I would like to hear about your response when I use this:, the number input I'm trying to use is for phone number

```tsx
<NumberInput
          hideControls
          mt="md"
          type="number"
          placeholder="Masukkan No. Handphone"
          {...props.form.getInputProps('phone_number')}
        />
```

do I need t handle the `onChange` myself?

**Reply by rtivital** [reply]
Yes, you need to add onChange prop on your side

**Reply by rimzzlabs** [reply]
Hi, sorry for late reply
I think you might misunderstood my question.
what I meant here is, every time I type number contain `0` value, example: `089412`, in number input, and then lose focus, the `0` value is removed, how to disable that behavior? 

P.S: currently our team using version 6 of mantine
